### ISSUE TITLE
Remove warnings and notices

### ISSUE THREAD

**Issue by andremacola** [ampproject/amp-wp#2154]
How to remove all the warnings and notices from the post edit page? Classical and Block editor.

**Comment by amedina** [comment]
Which notices are you referring to? Please provide screenshots or other pointers.

**Comment by westonruter** [comment]
If referring to the AMP validation error warnings, the way to hide them is to accept the validation errors.

@amedina We may want to make that explicit in the warning. I realize that we may not explain how to get rid of that warning. Currently it says:

> There is 1 issue from AMP validation which needs review. 1 issue is directly due to content here. However, your site is configured to automatically accept sanitization of the offending markup. **Review issues**

This could be amended with more details.

@andremacola Please confirm the notices you are referring to.

**Comment by andremacola** [comment]
@westonruter yes this is the notices.

I work for a relative big newspaper and this notices is annoying for the journalists. I think they need to appear (in our case) only for developers and the main editor.

We had some hard-working to migrate all of our system to Gutenberg, the journalists (specially the old ones) don't care about blocks and this kind of technical notice, they just want to write their text.

What I am doing right now is using get_amp_validation_error action to return false on post screens, but I do not known if this is the right solution. I miss some kind in-depth documentation.

![Captura de Tela 2019-04-21 às 16 39 49](https://user-images.githubusercontent.com/3408809/56474730-28707580-6454-11e9-9c3b-b772477f5cb6.png)

**Comment by westonruter** [comment]
Wouldn't it be important for the writers to know that some of the blocks they are authoring are going to be stripped from the AMP page? It seems an important tool for them to report a block problem to you.

Also, what are the validation errors you are seeing? Can't you mark the validation errors as accepted and then the writers won't see the warnings anymore?

**Comment by andremacola** [comment]
Not all the writers. Just the editors.

One of the notices is the video embed from the other ticket and some of our own stuff that for now is not amp compatible.

**Comment by westonruter** [comment]
OK, I think if we can fix the problem with #2156 then I can share a couple other filters which will reduce the noise for the validation errors that you are getting. Namely, the `amp_validation_error` filter can be used to normalize the validation errors (to reduce duplicates) and the `amp_validation_error_sanitized` filter can be used to programmatically accept sanitization and thus bypass the warning notice.

**Comment by andremacola** [comment]
Nice. `get_amp_validation_error` hook is solving the problem for now. It's time to translate the plugin to my language now.

Thks for the help @westonruter

**Comment by westonruter** [comment]
@andremacola You mean the `amp_validation_error` hook, right?

Would you mind sharing examples of how you are using it? This would be helpful to know how useful it is for developers.

**Comment by andremacola** [comment]
@westonruter actually is `get_amp_validation_error`. I do not remember where I saw this, I think I got from a global `wp_filter`

It's very simple by the way. It's like:

```
add_action('get_amp_validation_error', 'onyx_amp_disable_notices', 10, 1);
function onyx_amp_disable_notices($args) {
	global $typenow;
	if ( $typenow == 'post' && current_user_can('manage_options') ) {
		return false;
	} else {
		return $args;
	}
}
```

I'll check the `amp_validation_error`

**Comment by westonruter** [comment]
@andremacola I don't see `get_amp_validation_error` anywhere in the codebase. I don't see how your code snippet would work. 

The intended usage of the `amp_validation_error` filter is like this:

Given you have some HTML like this:

```html
<iframe src="https://example.com/video/12345" onload="Foo.loaded(this)"></iframe>
```

Consider `12345` being a video ID that is unique. This would have a validation error that looks like this:

![image](https://user-images.githubusercontent.com/134745/56483563-8797af80-647f-11e9-854e-3f4edb13fff6.png)

Now consider you have thousands such iframes, where the `src` is different for each. Without doing anything, there would be thousands of unique validation errors because a validation error includes the attributes of the invalid element. For example:

```json
{
    "element_attributes": {
        "src": "https://example.com/video/12345",
        "onload": "Foo.loaded(this)",
        "sandbox": "allow-scripts allow-same-origin",
        "height": "400",
        "layout": "fixed-height"
    },
    "node_name": "onload",
    "parent_name": "amp-iframe",
    "code": "invalid_attribute",
    "type": "js_error"
}
```

So what is needed is to normalize the error across all instances. And this can be done in this example as follows:

```php
add_filter(
	'amp_validation_error',
	function ( $error ) {
		$video_base_url  = 'https://example.com/video/';
		$is_video_iframe = (
			isset( $error['node_name'], $error['parent_name'], $error['element_attributes']['src'] )
			&&
			'onload' === $error['node_name']
			&&
			'amp-iframe' === $error['parent_name']
			&&
			substr( $error['element_attributes']['src'], 0, strlen( $video_base_url ) ) === $video_base_url
		);
		if ( $is_video_iframe ) {
			// Normalize URL across validation errors.
			$error['element_attributes']['src'] = $video_base_url . '{id}';
		}
		return $error;
	}
);
``` 

Now with this in place, every validation error for every such video `iframe` will look like this instead:

![image](https://user-images.githubusercontent.com/134745/56483810-f6c1d380-6480-11e9-82b0-374624439f43.png)

And now you can just accept this one validation error _once_ and every other instance of the validation error will have that same status, and the writer will not be shown a validation error in the editor.

This is the intended usage of the `amp_validation_error` filter.

**Comment by andremacola** [comment]
hey @westonruter I known, I searched inside the plugin files and did not find the `get_amp_validation_error`. I really do not remember where I saw this. I think I was debugging wordpress filters and hooks inside admin and I saw this.

![hook](https://user-images.githubusercontent.com/3408809/56500865-e2162780-64e2-11e9-8797-ad29fa2d3bc5.gif)

I'll try to find again when the time permit.

I just want to disable all kind of warnings from some kind os users, and believe, the simple little hook it just works for me.
